conans/test/functional/toolchains/cmake/test_ninja.py

```diff
@@ -1,144 +1,127 @@
 import textwrap
 import unittest
 import platform
-
+import os
 import pytest
 
+from conans.client.tools.files import which
 from conan.tools.microsoft.visual import vcvars_command
-from conans.test.assets.sources import gen_function_cpp
-from conans.test.functional.utils import check_vs_runtime
+from conan.tools.cmake.base import CMakeToolchainBase
+from conans.test.functional.utils import check_vs_runtime, check_msvc_library
 from conans.test.utils.tools import TestClient
-from conans.client.tools import which
+from conans.test.functional.toolchains.ios._utils import create_library
 
 
 @pytest.mark.tool_cmake
-class CMakeNinjaTestCase(unittest.TestCase):
+class TestCMakeNinja:
     # This test assumes that 'CMake' and 'Ninja' are available in the system
 
-    main_cpp = gen_function_cpp(name="main")
-    cmake = textwrap.dedent("""
-        cmake_minimum_required(VERSION 3.15)
-        project(App CXX)
-        set(CMAKE_VERBOSE_MAKEFILE ON)
-        add_executable(App main.cpp)
-        install(TARGETS App RUNTIME DESTINATION bin)
-        """)
     conanfile = textwrap.dedent("""
         from conans import ConanFile
         from conan.tools.cmake import CMake, CMakeToolchain
 
-        class Foobar(ConanFile):
-            name = "foobar"
-            settings = "os", "arch", "compiler", "build_type"
-            exports_sources = "CMakeLists.txt", "main.cpp"
+        class Library(ConanFile):
+            name = 'hello'
+            version = '1.0'
+            settings = 'os', 'arch', 'compiler', 'build_type'
+            exports_sources = 'hello.h', 'hello.cpp', 'CMakeLists.txt'
+            options = {'shared': [True, False]}
+            default_options = {'shared': False}
+            _cmake = None
+
+            def _configure_cmake(self):
+                if not self._cmake:
+                    self._cmake = CMake(self, generator="Ninja", parallel=False)
+                    self._cmake.configure()
+                return self._cmake
 
             def generate(self):
-                tc = CMakeToolchain(self, generator="Ninja")
+                tc = CMakeToolchain(self)
                 tc.generate()
 
             def build(self):
-                cmake = CMake(self)
+                cmake = self._configure_cmake()
                 cmake.configure()
                 cmake.build()
 
             def package(self):
-                cmake = CMake(self)
-                cmake.configure()
+                cmake = self._configure_cmake()
                 cmake.install()
         """)
 
     @classmethod
-    def setUpClass(cls):
+    def setup_class(cls):
         if not which("ninja"):
             raise unittest.SkipTest("Ninja expected in PATH")
 
-    @pytest.mark.skip("Not tested yet")
-    def test_locally_build_linux(self):
+    def setup_method(self):
```

Comment:
Not for now, but for future tests, better check pytest fixtures, they are quite nice.

Reply:
Done